### ISSUE TITLE
ROCm jobs are failing with No space left on device 

### ISSUE THREAD

**Issue by kit1980** [pytorch/pytorch#99578]
## Current Status
mitigated

## Error looks like
Error: No space left on device : '/home/pytorchci/actions-runner/_diag/Worker_20230419-234143-utc.log'
https://github.com/pytorch/pytorch/actions/runs/4748583207/jobs/8435235814

## Incident timeline (all times pacific)
*Include when the incident began, when it was detected, mitigated, root caused, and finally closed.*

<details>
<summary> Click for example </summary>

e.g.
- 10/30 7:27a incident began
- 10/30 8:30a detected by <method>
- 10/30 9:00 pm root caused as…
- 10/30 9:10 pm mitigated by…
- 10/31 10: am closed by…

</details>

## User impact
*How does this affect users of PyTorch CI?*

## Root cause
*What was the root cause of this issue?*

## Mitigation
Notified AMD via Slack.
Temporarily moving the jobs from trunk to "unstable".

## Prevention/followups
*How do we prevent issues like this in the future?*

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport

**Comment by jithunnair-amd** [comment]
Cleared docker images from worker-rocm-amd-108. 
Before:
```
pytorchci@jenkins-worker-rocm-amd-108:~$ docker images
REPOSITORY                                                                            TAG                                        IMAGE ID       CREATED        SIZE
308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-focal-rocm-n-py3   75cb70fd48f37e1e4b865493439717a7d6fe7f78   00e1a349ae83   6 days ago     35.8GB
308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-focal-rocm-n-py3   96e43b6d32395f7a02efef54e02053a907982e37   1e8d29614e55   8 days ago     35.4GB
308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-focal-rocm-n-py3   b43789f79f5fc01cfab4e4c785ad0fe29b90e0c1   29603ed2197d   8 days ago     35.8GB
308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-focal-rocm-n-py3   35de2104aba41e886c262ae07365a932c917b0d5   ec04d33e4ad0   2 weeks ago    26.4GB
pytorch/libtorch-cxx11-builder                                                        rocm5.4.2                                  0ece2dfab1da   2 weeks ago    19.3GB
pytorch/manylinux-builder                                                             rocm5.3                                    aee3b9c3c5c7   3 weeks ago    20.4GB
308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-focal-rocm-n-py3   51c670a3d430f7c7d9886677c30a0f7ce33e23ab   59c008315c5a   3 weeks ago    26.4GB
308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-focal-rocm-n-py3   f7fc29a4e736b46d6bb91df706930b434e68858f   fff17684903d   3 weeks ago    26.5GB
pytorch/manylinux-builder                                                             rocm5.4.2                                  5f1644853b4e   3 weeks ago    21.2GB
308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-focal-rocm-n-py3   31900f399b272261cdf50a84ee93cd7a0510c468   50d8f591e374   3 weeks ago    26.5GB
308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-focal-rocm-n-py3   476970eaf59e48d33e5db26305f233be3d9079b4   5cd973866e0b   3 weeks ago    26.5GB
308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-focal-rocm-n-py3   a415d9e2b28e9d1558d7c20de7a012c00c6c634c   2c82fb8e3e9c   4 weeks ago    26.2GB
308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-focal-rocm-n-py3   0d2d4165826ad2b39e6a0502d500cf5f7c3342aa   cdc75f179252   4 weeks ago    26.2GB
pytorch/libtorch-cxx11-builder                                                        rocm5.3                                    d63f397c3f32   4 weeks ago    18.5GB
308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-focal-rocm-n-py3   0fed3694b94c1fcbff2724f7badd2483c82a8a69   e41559c7bbd1   4 weeks ago    26.2GB
308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-focal-rocm-n-py3   a9899c47f90ad6213c1c42176655b1f7f2ac3a63   2a67262d65b1   4 weeks ago    26.2GB
308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-focal-rocm-n-py3   9ae4066a364a24ae2aae415da2a9f63765ce9163   5c606b79b19c   5 weeks ago    26.2GB
308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-focal-rocm-n-py3   99968ec69a116ed14b8105084bf99fb1edc98d69   55eeccd21ea0   5 weeks ago    25.9GB
308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-focal-rocm-n-py3   fadf10ce2391ba6c62d3bef60b68f92b635268f7   c6bcb51b9ef2   6 weeks ago    25.9GB
308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-focal-rocm-n-py3   4adb05210eff31b7fe4e1c7ed867f855e7a80701   509e430236f5   2 months ago   25.9GB
```

After:
```
pytorchci@jenkins-worker-rocm-amd-108:~$ docker images
REPOSITORY                                                                            TAG                                        IMAGE ID       CREATED      SIZE
308535385114.dkr.ecr.us-east-1.amazonaws.com/pytorch/pytorch-linux-focal-rocm-n-py3   75cb70fd48f37e1e4b865493439717a7d6fe7f78   00e1a349ae83   6 days ago   35.8GB
```

**Comment by kit1980** [comment]
I'll move ROCm back to trunk and close the issue after making sure the jobs are passing.

**Comment by huydhn** [comment]
@jithunnair-amd I have the list of the last 50 jobs executed by `worker-rocm-amd-108` here FYI.  They are sort by time https://paste.sh/_uEfvhgk#kHxgAj_ZMc6lFJbYXl3eUbEl

**Comment by jithunnair-amd** [comment]
> @jithunnair-amd I have the list of the last 50 jobs executed by `worker-rocm-amd-108` here FYI. They are sort by time https://paste.sh/_uEfvhgk#kHxgAj_ZMc6lFJbYXl3eUbEl
> 
> I could provide more if you need them.

@huydhn Thanks a ton, that's super helpful! Here's the culprit:
```
{
    "name": "linux-focal-rocm5.4.2-py3.8 / test (distributed, 1, 2, linux.rocm.gpu)",
    "html_url": "https://github.com/pytorch/pytorch/actions/runs/4746152574/jobs/8429935833",
    "_event_time": "2023-04-19T17:52:16.103975Z",
    "id": 12872794839,
    "run_id": 4746152574,
    "head_branch": "ciflow/unstable/97988",
    "head_sha": "7308aa3f7c039e8f011c9bf562a270e24e41e9e6",
    "workflow_name": "unstable-periodic"
  },
```

**Comment by kit1980** [comment]
I see this is no longer happening, moving the jobs back to trunk from unstable and closing this.

**Comment by huydhn** [comment]
There is another one running out of space today https://github.com/pytorch/pytorch/actions/runs/4757097717/jobs/8453943527 `worker-rocm-amd-106`

**Comment by jithunnair-amd** [comment]
> There is another one running out of space today https://github.com/pytorch/pytorch/actions/runs/4757097717/jobs/8453943527 `worker-rocm-amd-106` cc @jithunnair-amd

I'm going to clean up space on the machine right now. Would you kindly send me the history of jobs on that machine like you did for the other one? I have a hunch it's the same PR causing it, but would like to confirm.

**Comment by huydhn** [comment]
Here you go https://paste.sh/f5vVIgia#WlrBqXCyqN8T6kb7edyLrpQK for `worker-rocm-amd-106`

**Comment by jithunnair-amd** [comment]
> Here you go https://paste.sh/f5vVIgia#WlrBqXCyqN8T6kb7edyLrpQK for `worker-rocm-amd-106`

It's the same culprit PR again:
```
{
    "name": "linux-focal-rocm5.4.2-py3.8 / test (distributed, 2, 2, linux.rocm.gpu)",
    "html_url": "https://github.com/pytorch/pytorch/actions/runs/4755230044/jobs/8451106474",
    "_event_time": "2023-04-20T15:25:39.861985Z",
    "runner_name": "worker-rocm-amd-106",
    "head_branch": "ciflow/unstable/97988",
    "head_sha": "c15118fde82ad4b764b42f5bc82c50bd6942f144",
    "workflow_name": "unstable-periodic",
    "conclusion": "skipped",
    "line": null
  },
```

I have asked the PR author to not trigger any more CI runs until we figure out why that PR is resulting in artifacts that take up the entire disk space.

**Comment by huydhn** [comment]
I wonder if there is a way to detach the workflow from what PR.  Removing `ciflow/unstable` doesn't help once is has been attached.  Let me check

**Comment by huydhn** [comment]
Btw, I think the one you fixed yesterday `worker-rocm-amd-108` is now filling up again.  I will ping you the Rockset query to get the data on Slack

**Comment by kit1980** [comment]
I moved to unstable again as it still failing currently.

**Comment by jithunnair-amd** [comment]
> I wonder if there is a way to detach the workflow from the PR. Removing `ciflow/unstable` or `ciflow/trunk` doesn't help once it has been attached. Let me check

Wouldn't a rebase help even without removing the label?

**Comment by wbigat** [comment]
There is another one running out of space today,  what can i do to fix this?    @jithunnair-amd
https://github.com/pytorch/pytorch/pull/99712

**Comment by isdanni** [comment]
#99771 got one failed test with`Process exited with code 134`.

**Comment by jithunnair-amd** [comment]
> There is another one running out of space today, what can i do to fix this? @jithunnair-amd #99712

I'm sorry if I'm missing something, but I didn't find the failing run on https://hud.pytorch.org/pr/99712. Can you please provide a link to the failed Github Action job?

**Comment by jithunnair-amd** [comment]
> #99771 got one failed test with`Process exited with code 134`.

I see this on https://hud.pytorch.org/pr/99771:
```
X cuda11.8-py3.10-gcc7-sm80 / test (inductor_torchbench_smoketest_perf, 1, 1, linux.gcp.a100)
Raw logs | more like this | Queued: 1s | Duration: 23m 38s | Started: Sat 2:09 am | Test insights
▶ ##[error]Process completed with exit code 134.
```
This is not a ROCm job.
